### ISSUE TITLE
Should clean before building project when build flags change

### ISSUE THREAD

**Issue by 12qu** [amperka/ino#112]
Since we're now able to change features of the build (e.g. defined constants and the build tools themselves), build errors can occur when we change these features and don't clean before rebuilding.

For example, if I:
1. Specify --cflags=-DFOO=1
2. Run 'ino build'
3. Change to --cflags=-DFOO=2
4. Rerun 'ino build',

then the build in 4. will still be using 1 as the value for FOO, since these flags are hardcoded into the Makefiles in .build, which have not changed.

This can be fixed by making ino keep track of the last set of build flags and running clean immediately before the next if any of them have changed.

**Comment by nkrkv** [comment]
Agree. I think build flags should be put in some kind of hash that is used to separate builds the same way as it done for board model.
